doc/getting_started/index.rst

```diff
@@ -3,319 +3,372 @@
-configuration to work properly.
+   .. tabs::
 
-See :ref:`application_run` in the Application Development Primer and your
-board's documentation in :ref:`boards` for additional details.
+      .. group-tab:: Linux
 
-Run the Application in QEMU
-===========================
+         .. code-block:: bash
 
-On Linux and macOS, you can run Zephyr applications in emulation on your host
-system using QEMU when targeting either the x86 or ARM Cortex-M3 architectures.
+            cd ~/zephyrproject/zephyr
+            source zephyr-env.sh
 
-To build and run Hello World using the x86 emulation board configuration
-(``qemu_x86``), type:
+      .. group-tab:: macOS
 
-.. zephyr-app-commands::
-   :zephyr-app: samples/hello_world
-   :host-os: unix
-   :board: qemu_x86
-   :goals: build run
+         .. code-block:: bash
 
-To exit, type :kbd:`Ctrl-a`, then :kbd:`x`.
+            cd ~/zephyrproject/zephyr
+            source zephyr-env.sh
 
-Use ``qemu_cortex_m3`` to target an emulated Arm Cortex-M3 instead.
+      .. group-tab:: Windows
 
-.. _gs_posix:
+         .. code-block:: bash
 
-Run a Sample Application natively (POSIX OS)
-============================================
+            cd %HOMEPATH%/zephyrproject/zephyr
+            zephyr-env.cmd
 
-Finally, it is also possible to compile some samples to run as host processes
-on a POSIX OS. This is currently only tested on Linux hosts. See
-:ref:`native_posix` for more information. On 64 bit host operating systems, you
-need to install a 32 bit C library; see :ref:`native_posix_deps` for details.
+#. Build the blinky sample. Specify **your board** in the command below.
+   This command builds blinky for the :ref:`Phytec reel board <reel_board>`:
 
-First, build Hello World for ``native_posix``.
+   .. zephyr-app-commands::
+      :tool: west
+      :app: samples/basic/blinky
+      :board: reel_board
+      :goals: build
+
+
+Run the Application by Flashing to a Board
+==========================================
 
-.. zephyr-app-commands::
-   :zephyr-app: samples/hello_world
-   :host-os: unix
-   :board: native_posix
-   :goals: build
+#. Connect a USB cable between the board and your development computer.
+   (Refer to the specific :ref:`boards` documentation if you're not sure
+   which connector to use on the board.)
+#. Turn it on (if there's a switch).
+#. Flash the blinky application you just built using the commmand below:
 
-Next, run the application.
+.. tabs::
 
-.. code-block:: console
+   .. group-tab:: Linux
 
-   west build -t run
-   # or just run zephyr.exe directly:
-   ./build/zephyr/zephyr.exe
+      Flashing the board requires permission to directly access
+      the hardware.  The simple solution is to use ``sudo``, as shown here:
 
-Press :kbd:`Ctrl-C` to exit.
+      .. code-block:: bash
 
-You can run ``./build/zephyr/zephyr.exe --help`` to get a list of available
-options.
+         sudo -E PATH=$PATH sh -c "west flash"
```

Comment:
> But if you add sudo, the newly generated files can be owned by root. 

Building as root is a big no-no. Especially not after pip has downloaded a bunch of random packages from the Internet, half of them users don't need (I understand different users need the other half)

> I think we cannot paper over the inherent complexity of device permissions on linux here, and ought to add information about necessary udev rules on supported platforms to the documentation as a troubleshooting step if this doesn't work.

Unfortunately agreed 100%. There's a lot that good automation and good documentation can achieve, but they can solve neither world hunger nor the complicated and fragmented state of udev security across Linux distributions. The best that can be done here is:
1. Make sure Ubuntu 12.34 works
2. For the 1000s of other Linux distributions, defer. Provide generic _test_ instructions or even better: generic test code that prints: "You don't seem to have permission to access to /dev/whatever. Please check the udev part of your Linux distribution"

BTW it would _really_ be nice for chocolatey to run without the Adminstrator account. I digress.

Reply:
I stand by the discussion from earlier. For whatever it's worth, flashing as root in the GSG is not OK with me.

If your flash tool installer doesn't set up udev rules properly, its installer is broken.

(Nordics' tools installed via .deb do take care of this properly, e.g.)

Reply:
I'm really, really not OK with flashing as root. I don't like that this was marked resolved without any changes.

Reply:
@mbolivar Do you have a counter proposal that would generically work for some set of supported boards?  I'm trying to find a simple path for a new developer to get a first sample app running on a board.  I tried saying using sudo is not the proper way, but we'll use it for this quick start. Let's talk offline and figure this out.

Reply:
Ignoring the security disaster for a moment, a build directory that can't be cleaned because it has a random number of files owned by root is everything but new user friendly!

> Do you have a counter proposal that would generically work for some set of supported boards?

A generic solution that works across a wide range of boards and Linux distros is probably impossible. On the other hand, a single udev rule that works reliably on one specific Ubuntu version for a reasonable number of boards shouldn't be rocket science.

Breaking news: security is rarely user-friendly.

Reply:
Talking with the bug triage folks, they recommend removing the ``sudo`` and just do ``west flash`` and add a troubleshooting note about udev rules.  As you mention, security isn't free.

Reply:
>  Talking with the bug triage folks, they recommend removing the sudo and just do west flash and add a troubleshooting note about udev rules. As you mention, security isn't free.

That's reasonable. Thanks.

Reply:
> a build directory that can't be cleaned because it has a random number of files owned by root is everything but new user friendly!

^^ also, it's not just a security thing. This is also an issue.